pkg/pillar/cmd/volumemgr/handlevolume.go

```diff
@@ -192,6 +193,26 @@ func publishVolumeStatus(ctx *volumemgrContext,
 	status *types.VolumeStatus) {
 
 	key := status.Key()
+	if ctx.hvTypeKube {
+		vrStatus := lookupVolumeRefStatus(ctx, key)
+		sub := ctx.pubContentTreeStatus
+		items := sub.GetAll()
+		reference := ""
+		for _, item := range items {
+			cts := item.(types.ContentTreeStatus)
+			if status.ContentID.String() == cts.ContentID.String() && cts.IsContainer() {
+				reference = cts.ReferenceID()
+				break
+			}
+		}
+		if vrStatus != nil && reference != "" {
+			if !strings.HasPrefix(vrStatus.ReferenceName, types.KubeContainerImagePrefix) && vrStatus.IsContainer() {
+				vrStatus.ReferenceName = types.KubeContainerImagePrefix + reference
+				log.Functionf("publishVolumeStatus: sync reference name %s", vrStatus.ReferenceName)
+				publishVolumeRefStatus(ctx, vrStatus)
+			}
+		}
+	}
```

Comment:
Why do these extra publish belong here?
volumemgr has code which makes sure that a change to a VolumeStatus is reflected to all the VolumeRefStatus which depend on it. Is that code not working for kubevirt? Not being invoked?
Or is this leftover code which is not needed any more?